Source/utils/str_split.hpp

```diff
@@ -0,0 +1,98 @@
+#pragma once
+
+#include <iterator>
+#include <type_traits>
+
+#include "stdcompat/string_view.hpp"
+
+namespace devilution {
+class SplitByCharIterator {
+public:
+	using iterator_category = std::forward_iterator_tag;
+	using value_type = string_view;
+	using reference = std::add_lvalue_reference<value_type>::type;
+	using pointer = std::add_pointer<value_type>::type;
+
+	static SplitByCharIterator begin(string_view text, char split_by) // NOLINT(readability-identifier-naming)
+	{
+		return SplitByCharIterator(split_by, text, text.substr(0, text.find(split_by)));
+	}
+
+	static SplitByCharIterator end(string_view text, char split_by) // NOLINT(readability-identifier-naming)
+	{
+		return SplitByCharIterator(split_by, text, text.substr(text.size()));
+	}
+
+	[[nodiscard]] string_view operator*() const
+	{
+		return slice_;
+	}
+
+	[[nodiscard]] const string_view *operator->() const
+	{
+		return &slice_;
+	}
+
+	SplitByCharIterator &operator++()
+	{
+		slice_ = text_.substr(slice_.data() - text_.data() + slice_.size());
+		if (!slice_.empty())
+			slice_.remove_prefix(1); // skip the split_by char
+		slice_ = slice_.substr(0, slice_.find(split_by_));
+		return *this;
+	}
+
+	SplitByCharIterator operator++(int) const
+	{
+		auto copy = *this;
+		++copy;
+		return copy;
+	}

```

Comment:
I'm not 100% sure, but this feels wrong.

```suggestion
	SplitByCharIterator operator++(int) const
	{
		auto copy = *this;
		++*this;
		return copy;
	}
```

Reply:
Fixed!